### PR TITLE
feat: Restore acronym and acronym-from-file CLI flags

### PR DESCRIPTION
**Pull request by arran4**
This PR resolves the issue of missing `-acronym` and `-acronym-from-file` CLI flags which were lost when migrating off `iancoleman/strcase`.

It implements:
* A `mappers.MapAcronyms(acronyms ...string)` function that acts as a `WordMapper`, parsing case-insensitive words and casting them into explicitly cased `AcronymWord` types.
* Additions to `cli/main.go` and `buildOpts` that parse `--acronym` and `--acronym-from-file` arguments and dynamically aggregate them into a single list before instantiating the `WordMapper`. 
* Regenerated CLI commands using `go generate ./...`.

---
*PR created automatically by Jules for task [4698834321828491132](https://jules.google.com/task/4698834321828491132) started by @arran4*

### CODE REVIEW

File: cli/main.go [hunk 1]
```diff
@@ -81,6 +82,29 @@ func buildOpts(delimiter string, screaming bool, whispering bool, firstUpper boo
...
+				os.Exit(1)
+			}
+			lines := strings.Split(string(content), "\n")
+			for _, l := range lines {
+				l = strings.TrimSpace(l)
+				if l != "" {
+					allAcronyms = append(allAcronyms, l)
+				}
+			}
+		}
```
[the remark this answers is not in the document]
Comment (arran4) [reply cli/main.go:110]: > medium
> 
> Instead of reading the entire file into memory with `os.ReadFile` and splitting it by `\n`, it is more efficient and idiomatic to read the file line-by-line using `bufio.Scanner`. This avoids unnecessary allocations and handles different line endings (like `\r\n`) cleanly. Note that you will need to import `"bufio"` in this file.
> Suggested change
> 		for _, f := range acronymFromFile {
> 			content, err := os.ReadFile(f)
> 			if err != nil {
> 				fmt.Fprintf(os.Stderr, "Error loading acronym file %s: %v\n", f, err)
> 				os.Exit(1)
> 			}
> 			lines := strings.Split(string(content), "\n")
> 			for _, l := range lines {
> 				l = strings.TrimSpace(l)
> 				if l != "" {
> 					allAcronyms = append(allAcronyms, l)
> 				}
> 			}
> 		}
> 		for _, f := range acronymFromFile {
> 			file, err := os.Open(f)
> 			if err != nil {
> 				fmt.Fprintf(os.Stderr, "Error loading acronym file %s: %v\n", f, err)
> 				os.Exit(1)
> 			}
> 			scanner := bufio.NewScanner(file)
> 			for scanner.Scan() {
> 				l := strings.TrimSpace(scanner.Text())
> 				if l != "" {
> 					allAcronyms = append(allAcronyms, l)
> 				}
> 			}
> 			if err := scanner.Err(); err != nil {
> 				fmt.Fprintf(os.Stderr, "Error reading acronym file %s: %v\n", f, err)
> 				file.Close()
> 				os.Exit(1)
> 			}
> 			file.Close()
> 		}

@jules

File: mappers/acronyms.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+				continue
+			}
+			s := w.String()
+			lowerS := strings.ToLower(s)
+			if replacement, ok := acronymMap[lowerS]; ok {
+				result = append(result, strings2.AcronymWord(replacement))
+			} else {
+				result = append(result, w)
+			}
+		}
```
[the remark this answers is not in the document]
Comment (arran4) [reply mappers/acronyms.go]: > medium
> 
> To improve performance and correctness, we should pre-allocate the `result` slice with the capacity of `words` to avoid multiple allocations during `append`. Additionally, we should preserve `nil` elements in the slice instead of silently filtering them out, which could unexpectedly alter the structure of the word slice.
> Suggested change
> 		var result []strings2.Word
> 		for _, w := range words {
> 			if w == nil {
> 				continue
> 			}
> 			s := w.String()
> 			lowerS := strings.ToLower(s)
> 			if replacement, ok := acronymMap[lowerS]; ok {
> 				result = append(result, strings2.AcronymWord(replacement))
> 			} else {
> 				result = append(result, w)
> 			}
> 		}
> 		result := make([]strings2.Word, 0, len(words))
> 		for _, w := range words {
> 			if w == nil {
> 				result = append(result, nil)
> 				continue
> 			}
> 			s := w.String()
> 			lowerS := strings.ToLower(s)
> 			if replacement, ok := acronymMap[lowerS]; ok {
> 				result = append(result, strings2.AcronymWord(replacement))
> 			} else {
> 				result = append(result, w)
> 			}
> 		}

@jules

File: mappers/acronyms_test.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+	expected := []strings2.Word{
+		strings2.SingleCaseWord("user"),
+		strings2.AcronymWord("ID"),
+	}
+
+	if !reflect.DeepEqual(result, expected) {
+		t.Errorf("Expected %v, got %v", expected, result)
+	}
+}
```
[the remark this answers is not in the document]
Comment (arran4) [reply mappers/acronyms_test.go:72]: > medium
> 
> Please add unit tests for the new `MapAcronymsFromFile` function to ensure it correctly reads acronyms from a file and handles errors (e.g., file not found). Additionally, consider adding a test case to `TestMapAcronyms` that verifies `nil` words are safely preserved without causing panics or being filtered out.

@jules

File: mappers/acronyms.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+		return result
+	}
+}
+
+// MapAcronymsFromFile reads a file containing one acronym per line and returns a WordMapper.
```
Comment (arran4) [inline mappers/acronyms.go]: @jules we also need a io.Reader and []byte version of this as well as a bunch of sample files with a go:embed publically exported version in their own package.

File: mappers/samples/samples.go [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package samples
+
+import _ "embed"
+
+//go:embed tech.txt
+var TechAcronyms []byte
```
Comment (arran4) [inline mappers/samples/samples.go:10]: @jules produce a reader function specifically for this data

File: mappers/samples/tech.txt
Comment (arran4) [inline mappers/samples/tech.txt:1]: @jules bigger sample and more sample files too please.

File: mappers/acronyms.go [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+	return MapAcronyms(acronyms...), nil
+}
+
+// MapAcronymsFromBytes reads acronyms from a []byte (one per line) and returns a WordMapper.
+func MapAcronymsFromBytes(b []byte) (func([]strings2.Word) []strings2.Word, error) {
+	return MapAcronymsFromReader(bytes.NewReader(b))
+}
+
+// MapAcronymsFromFile reads a file containing one acronym per line and returns a WordMapper.
+func MapAcronymsFromFile(filename string) (func([]strings2.Word) []strings2.Word, error) {
```
Comment (arran4) [inline mappers/acronyms.go:68]: @jules provide a URL feeder from some CDN'd txt file. If one doesn't exist we can make it part of the github pages of this project, using the read me and https://github.com/arran4/fork-mdtohtml/releases/latest in a github action (action deploy not branch.)

File: mappers/acronyms_test.go [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+func TestMapAcronymsFromURL(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte("URL\nHTTP\n"))
```
Comment (arran4) [inline mappers/acronyms_test.go]: @jules Error return value of `w.Write` is not checked (errcheck)
